app/components/login.js

```diff
@@ -0,0 +1,33 @@
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { AUTH } from '../constants/urls';
+import { inject as service } from '@ember/service';
+
+export default class LoginComponent extends Component {
+  @service router;
+  @service fastboot;
+  @service login;
+  @tracked authURL = this.generateAuthURL();
+
+  AUTH_URL = this.generateAuthURL();
+
+  generateAuthURL() {
+    let currentURL = this.fastboot.isFastBoot
+      ? this.fastboot.request.protocol +
+        '//' +
+        this.fastboot.request.host +
+        this.fastboot.request.path
+      : window.location.href;
```

Comment:
Does it even happen?

Reply:
it does, it generates the auth url

Reply:
Please explain, what is happening here
Explain the fastboot part please.

Reply:
this is snippet gets url of current route to pass it off in redirect query in auth route.
fastboot is required early during page load if window object is not available.